spec/factories/metadata/teacher_lead_provider_factory.rb

```diff
@@ -0,0 +1,53 @@
+FactoryBot.define do
+  factory(:teacher_lead_provider_metadata, class: "Metadata::TeacherLeadProvider") do
+    association :teacher
+    association :lead_provider
+
+    transient do
+      ect_teacher_type { false }
+      mentor_teacher_type { false }
+    end
+
+    after(:create) do |metadata, evaluator|
+      teacher = metadata.teacher
+      lead_provider = metadata.lead_provider
+
+      contract_period = FactoryBot.create(:contract_period, :current)
+      active_lead_provider = FactoryBot.create(:active_lead_provider, contract_period:, lead_provider:)
+      lead_provider_delivery_partnership = FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:)
+      school_partnership = FactoryBot.create(:school_partnership, lead_provider_delivery_partnership:)
+      school = school_partnership.school
+
+      [
+        (:ect if evaluator.ect_teacher_type),
```

Comment:
Would it be easier to pass in this as the value to `ect_teacher_type`? so `ect_teacher_type { :ect }` - I think we will want some teachers with both types as well so maybe supporting an array, `ect_teacher_types { %i[ect mentor] }` could be useful

Reply:
you can't pass in as you might want to do something like this:
```ruby
FactoryBot.create :teacher_lead_provider_metadata, :with_latest_ect_training_period, :with_latest_mentor_training_period
```

You want to use the trait to enable ect or mentor